packages/react-magma-dom/src/components/Input/index.tsx

```diff
@@ -54,11 +74,13 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
             descriptionId ? descriptionId : props['aria-describedby']
           }
           aria-invalid={!!errorMessage}
-          hasError={!!errorMessage}
+          hasError={!!errorMessage || characterLength > maxLength}
           iconPosition={iconPosition}
           id={id}
           inputSize={inputSize}
           isInverse={isInverse}
+          onChange={maxLength ? handleChange : props.onChange}
```

Comment:
I think we can handle this like this instead of having the ternary on this call:

https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/components/InputBase/index.tsx#L508

https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/components/InputBase/index.tsx#L477-L483